File: backend/app/models/album.rb
class Album < ApplicationRecord
    has_many :songs
    belongs_to :artist
    belongs_to :genre

    def artist_name 
        self.artist ? self.artist.name : nil
    end

    def artist_name=(name)
        self.artist = Artist.find_or_create_by(name: name)
    end

    def songs_attributes=(attributes_hash)
        binding.pry
        attributes_hash.each do |k, value|
            binding.pry
            song = Song.new(value)
            song.artist_id = self.artist_id
            song.save
            self.songs << song 
        end
    end


end

File: beats-me-client/src/album.js
class Album {
    constructor(id, name, artist, genre, imgURL, songs){
        this.id = id
        this.name = name
        this.artist = artist
        this.genre = genre
        this.imgURL = imgURL
        this.songs = []
        songs.forEach(song => this.songs.push(song))
        this.constructor.all.push(this)
    } 
    
    static all = []

}